orangecontrib/text/widgets/owcorpusviewer.py

```diff
@@ -153,20 +164,26 @@ def load_features(self):
         self.display_indices = []
         if self.corpus is not None:
             domain = self.corpus.domain
+            self.closeContext()
+
             self.search_features = list(filter_visible(chain(domain.variables, domain.metas)))
             self.display_features = list(filter_visible(chain(domain.variables, domain.metas)))
-            # FIXME: Select features based on ContextSetting
-            self.search_indices = list(range(len(self.search_features)))
-            self.display_indices = list(range(len(self.display_features)))
+            self.list_search_indices = list(range(len(self.search_features)))
+            self.list_display_indices = list(range(len(self.display_features)))
 
+            self.openContext(self.corpus)
+            self.search_indices = list(self.list_search_indices)
+            self.display_indices = list(self.list_display_indices)
             # Enable/disable tokens checkbox
             if not self.corpus.has_tokens():
                 self.show_tokens_checkbox.setCheckState(False)
             self.show_tokens_checkbox.setEnabled(self.corpus.has_tokens())
 
     def list_docs(self):
         """ List documents into the left scrolling area """
-        search_keyword = self.filter_input.text().strip('|')
+        if not self.corpus_docs:
```

Comment:
Why this if? `list_docs` seems to only be called from `refresh_search` which already has an if self.corpus?